source/common/http/http1/codec_impl.cc

```diff
@@ -411,8 +411,9 @@ int ServerConnectionImpl::onHeadersComplete(HeaderMapImplPtr&& headers) {
         http_parser_pause(&parser_, 1);
       }
 
-    } else if (parser_.method == http_method::HTTP_POST ||
-               parser_.method == http_method::HTTP_PUT) {
+    } else if ((parser_.method == http_method::HTTP_POST ||
+                parser_.method == http_method::HTTP_PUT) &&
+               headers->ContentLength() == nullptr) {
```

Comment:
I hear you.   If the risks to prod traffic are not worth it on your end I can back this part out.    It reminds me though - it's worth talking at some point about strategies we've used internally to mitigate high risk changes to not lose binary pushes, and see if any are worth incorporating into Envoy :-)